### PR TITLE
Make UsePeculiarVelocities = 1 the default.

### PR DESCRIPTION
**Pull request by sbird**
This replaces https://github.com/MP-Gadget/MP-Gadget/pull/300

Makes UsePeculiarVelocities = 1 the default, fixes the format conversion script and adds some comments on the velocity format.

### CODE REVIEW

File: libgadget/timestep.c [hunk 1]
```diff
@@ -314,15 +314,15 @@ do_the_short_range_kick(int i, inttime_t tistart, inttime_t tiend)
...
-         * FIXME: This should probably not be needed!*/
-        const double velfac = sqrt(All.cf.a3inv);
+         * on the gas velocity. This should rarely be hit.*/
+        const double velfac = 1./All.cf.a;
         double vv=0;
         for(j=0; j < 3; j++)
             vv += P[i].Vel[j] * P[i].Vel[j];
         vv = sqrt(vv);
 
         if(vv > All.MaxGasVel * velfac) {
```
Comment (rainwoodman) [inline libgadget/timestep.c]: Shouldn't this always be vv * ainv > All.MaxGasVel?
Then in parameters, MaxGasVel shall be defined in terms of peculiar velocity, a dx / dt and thus 3e5 in km/s.

File: libgadget/petaio.c [hunk 2]
```diff
@@ -481,7 +481,10 @@ petaio_read_header_internal(BigFile * bf) {
         endrun(0, "Failed to read required cosmology from IC header\n");
     }
 
-    /* Fall back to use a**2 * dx/dt if UsePeculiarVelocity is not set in IC */
+    /* If UsePeculiarVelocity = 1 then snapshots save to the velocity field the physical peculiar velocity.
+     * If UsePeculiarVelocity = 0 then snapshots save a * physical peculiar velocity in snapshots
```
Comment (rainwoodman) [inline libgadget/petaio.c]: I settled to explicitly write down whenever these are mentioned the exact formula 
- a dx / dt for 'physical peculiar velocity', 
- a^2 dx / dt for conjugate momentum (per unit mass internal integration quantity, BlueTides-i/ii) and
- a^{1/2} dx / dt for Gadget-1 velocity.

I think the formula makes conversion clear and avoids confusion.

File: tools/convert_bigfile_gadget_format3.py [hunk 3]
```diff
@@ -113,7 +116,14 @@ def write_hdf_file(bf, hdf5name, fnum, nfiles):
                 continue
             ptype = int(ptype)
             hname = names.get_hdf5_name(bname)
-            hdf5["PartType"+str(ptype)][hname] =  bf[block][startpart[ptype]:endpart[ptype]]
+            bfdata = bf[block][startpart[ptype]:endpart[ptype]]
+            #Velocities need units adjusted
```
Comment (rainwoodman) [inline tools/convert_bigfile_gadget_format3.py]: # Gadget-3 uses a^{1/2} dx / dt. 

if pecvel: # pecvel is a dx / dt.

else # MP-Gadget@Bluetides snapshot, a^2 dx / dt.

**Comment by rainwoodman** [comment]
Looks good! I request a few changes in comments.

File: libgadget/petaio.c (hunk 2 above)
Comment (sbird) [reply libgadget/petaio.c]: That is a good idea. Just to explicitly check that I have this right: x is comoving distance in the above, right?

File: libgadget/timestep.c (hunk 1 above)
Comment (sbird) [reply libgadget/timestep.c]: Oops, yes it should! Nicely caught.

**Pull request merged**